src/main/kotlin/snyk/container/YAMLImageExtractor.kt

```diff
@@ -39,7 +39,7 @@ object YAMLImageExtractor {
     fun extractFromFile(file: VirtualFile, project: Project): Set<KubernetesWorkloadImage> {
         val psiFile = findPsiFileIgnoringExceptions(file, project)
         if (psiFile == null || file.isDirectory || !file.isValid) {
```

Comment:
`file.isDirectory || !file.isValid` should be done before obtaining PsiFile and if true - such obtaining not needed